src/loaders/common/loader_helper.py

```diff
@@ -69,11 +75,60 @@ def init_genome_atrri_doc(kbase_collection, load_version, genome_id):
     # The '_key' field for the document should be generated by applying a hash function to a combination of the
     # 'kbase_collection', 'load_version', and 'genome_id' fields.
     doc = {
-        names.FLD_ARANGO_KEY: md5_string(f"{kbase_collection}_{load_version}_{genome_id}"),
+        names.FLD_ARANGO_KEY: md5_string(
+            f"{kbase_collection}_{load_version}_{genome_id}"
+        ),
         names.FLD_COLLECTION_ID: kbase_collection,
         names.FLD_LOAD_VERSION: load_version,
         names.FLD_GENOME_ATTRIBS_KBASE_GENOME_ID: genome_id,  # Begin with the input genome_id, though it may be altered by the calling script.
-        names.FLD_GENOME_ATTRIBS_MATCHES_SELECTIONS: []  # for saving matches and selections
+        names.FLD_GENOME_ATTRIBS_MATCHES_SELECTIONS: [],  # for saving matches and selections
     }
 
     return doc
+
+
+def get_token(token_filename):
+    """
+    Get token if a filename is provided at home dir.
+    """
+    try:
+        home_dir = os.path.expanduser("~")
+        with open(os.path.join(home_dir, "." + token_filename), "r") as f:
+            token = f.readline().strip()
+    except FileNotFoundError as e:
+        print(e.errno)
+    return token
+
+
+def start_podman_service(uid):
+    """
+    Start podman service. Used by workspace_downloader.py script.
+    """
+    # try:
+    #     check_output(["pidof", "podman"])
+    # except CalledProcessError:
+    #     print("No running podmans servies are detected. Start one now!")
+
+    # TODO find out the right way to check if a podman service is running
+    proc = subprocess.Popen(["podman", "system", "service", "-t", "0"])
+    os.environ["DOCKER_HOST"] = DOCKER_HOST.format(uid)
+    return proc
+
+
+def is_upaInfo_complete(output_dir, upa):
+    """
+    Check whether an UPA needs to be downloaded or not by loading the metadata file.
+    Make sure it has all the right keys.
+    """
+    fa_path = os.path.join(output_dir, upa, upa + ".fa")
+    meta_path = os.path.join(output_dir, upa, upa + ".meta")
+    if not os.path.exists(fa_path) or not os.path.exists(meta_path):
+        return False
+    try:
+        with open(meta_path, "r") as json_file:
+            data = json.load(json_file)
+    except OSError:
+        return False
+    if set(data.keys()) != set(META_KEYS):
```

Comment:
Do you mean?
``` 
if not set(data.keys()) < set(META_KEYS):
    return False
return True
```

Reply:
no. I think data.keys() should at least include META_KEYS. So when `set(data.keys()) < set(META_KEYS)` return False.

Reply:
what if data.keys() contains non META_KEYS? then it will skip set(data.keys()) < set(META_KEYS) logic and then return True. But it actually should return False, right?

Reply:
I think Tian's approach is the right way to go - we should assume that there may be other keys in the file that this program doesn't control, and therefore just check that the keys we expect exist.

Reply:
Updated logic: All meta keys (We expect exist) must be a subset of keys in the file.

Reply:
:+1: